base/pointer.jl

```diff
@@ -35,38 +35,65 @@ unsafe_convert{S,T}(::Type{Ptr{S}}, a::AbstractArray{T}) = convert(Ptr{S}, unsaf
 unsafe_convert{T}(::Type{Ptr{T}}, a::AbstractArray{T}) = error("conversion to pointer not defined for $(typeof(a))")
 
 # unsafe pointer to array conversions
-function pointer_to_array{T}(p::Ptr{T}, d::Integer, own::Bool=false)
+function unsafe_array_wrapper{T}(p::Ptr{T}, d::Integer, own::Bool=false)
     ccall(:jl_ptr_to_array_1d, Vector{T},
           (Any, Ptr{Void}, Csize_t, Cint), Array{T,1}, p, d, own)
 end
-function pointer_to_array{T,N}(p::Ptr{T}, dims::NTuple{N,Int}, own::Bool=false)
+function unsafe_array_wrapper{T,N}(p::Ptr{T}, dims::NTuple{N,Int}, own::Bool=false)
     ccall(:jl_ptr_to_array, Array{T,N}, (Any, Ptr{Void}, Any, Int32),
           Array{T,N}, p, dims, own)
 end
-function pointer_to_array{T,N}(p::Ptr{T}, dims::NTuple{N,Integer}, own::Bool=false)
+function unsafe_array_wrapper{T,N}(p::Ptr{T}, dims::NTuple{N,Integer}, own::Bool=false)
     i = 1
     for d in dims
         if !(0 <= d <= typemax(Int))
             throw(ArgumentError("Array dimension must be 0 ≤ dim ≤ $(typemax(Int)), got $d for dimension $i"))
         end
         i += 1
     end
-    pointer_to_array(p, convert(Tuple{Vararg{Int}}, dims), own)
+    unsafe_array_wrapper(p, convert(Tuple{Vararg{Int}}, dims), own)
 end
+"""
+    unsafe_array_wrapper(pointer, dims, own=false)
+
+Wrap a native pointer as a Julia Array object. The pointer element type determines the array
+element type. `own` optionally specifies whether Julia should take ownership of the memory,
+calling `free` on the pointer when the array is no longer referenced.
+
+This function is labelled "unsafe" because it will crash if `pointer` is not
+a valid memory address to data of the requested length.
+"""
+unsafe_array_wrapper
```

Comment:
Ah, I see.   (It would be good if this were explained in the manual.  There's an obscure comment about the `function end` syntax being "preferred" with no explanation.)